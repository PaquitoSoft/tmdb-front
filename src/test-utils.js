import React from 'react';
import { shallow, mount } from 'enzyme';
import { AppProvider } from './components/app-context/app-context';

const userIdMock = 'hskfjahshjdjskla';

const imagesConfigMock = {
	"baseUrl":"https://image.tmdb.org/t/p/",
	"backdropSizes":["w300","w780","w1280","original"],
	"logoSizes":["w45","w92","w154","w185","w300","w500","original"],
	"posterSizes":["w92","w154","w185","w342","w500","w780","original"],
	"profileSizes":["w45","w185","h632","original"],
	"stillSizes":["w92","w185","w300","original"]
};

const renderers = {
	'shallow': shallow,
	'mount': mount
};

function AppContextMock({ children, apiClientMock }) {
	return (
		<AppProvider 
			apiClient={apiClientMock} 
			userId={userIdMock} 
			imagesConfig={imagesConfigMock}
		>
			{children}
		</AppProvider>
	);
}

export function render(
	component, 
	options = { renderType: 'shallow' }
) {
	return renderers[options.renderType](component);
}

export function validateSnapshot(
	component, 
	options = { renderType: 'shallow' }
) {
	const wrapper = render(component, options);
	expect(wrapper).toMatchSnapshot();
	return wrapper;
}

export function renderInAppContext(
	component, 
	options = {}
) {
	const _options = {
		renderType: 'mount',
		apiClientQueryMock: jest.fn().mockImplementation(() => Promise.resolve({})),
		apiClientMutationMock: jest.fn().mockImplementation(() => Promise.resolve({})),
		...options
	};

	const apiClientMock = {
		sendQuery: _options.apiClientQueryMock,
		sendMutation: _options.apiClientMutationMock
	};
	
	const wrapper = render(
		(
			<AppContextMock apiClientMock={apiClientMock}>
				{component}
			</AppContextMock>
		),
		_options
	);

	return wrapper.find(AppProvider).childAt(0);
}

export const clone = object => JSON.parse(JSON.stringify(object));
