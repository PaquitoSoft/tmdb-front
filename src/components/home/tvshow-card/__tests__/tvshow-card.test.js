import React from 'react';
import { validateSnapshot } from '../../../../test-utils';

import TvShowCard from '../tvshow-card';

describe('TvShowCard', () => {

	const tvShowMockData = {
		"id": 68507,
		"name": "His Dark Materials",
		"backdropImagePath": "/9yKCJTOh9m3Lol2RY3kw99QPH6x.jpg",
		"posterPath": "/xOjRNnQw5hqR1EULJ2iHkGwJVA4.jpg",
		"votesAverage": 7.5,
		"isFavorite": false,
		"firstAirDate": "2019-11-03"
	};

	it('Should render TvShow Card', () => {
		validateSnapshot(<TvShowCard tvShow={tvShowMockData} />);
	});

});
