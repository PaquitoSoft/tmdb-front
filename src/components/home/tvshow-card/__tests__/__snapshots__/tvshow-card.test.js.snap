// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvShowCard Should render TvShow Card 1`] = `
<Link
  className="tvshow-card"
  to="/tvshow/68507"
>
  <div
    className="tvshow-card__poster"
  >
    <MediaImage
      alt="His Dark Materials"
      className="tvshow-card__poster-image"
      path="/9yKCJTOh9m3Lol2RY3kw99QPH6x.jpg"
      sizes="38vw"
      type="backdrop"
    />
  </div>
  <div
    className="tvshow-card__footer"
  >
    <RatingIcon
      className="tvshow-cart__rating"
      ratingValue={7.5}
      size="big"
    />
    <div
      className="tvshow-card__info"
    >
      <span
        className="tvshow-card__title"
      >
        His Dark Materials
      </span>
      <span
        className="tvshow-card__date"
      >
        Sun Nov 03 2019
      </span>
    </div>
    <FavoriteIcon
      className="tvshow-card__favorite-icon"
      isActive={false}
    />
  </div>
</Link>
`;
